featuretools/synthesis/deep_feature_synthesis.py

```diff
@@ -840,8 +841,6 @@ def feature_filter(f):
             wheres = list(self.where_clauses[child_dataframe.ww.name])
 
             for matching_input in matching_inputs:
-                if not check_stacking(agg_prim, matching_input):
-                    continue
```

Comment:
These lines should probably stay, it's this removal that is causing the discrepancy in number of features generated

Reply:
Ah, good catch!